### ISSUE TITLE
NSGenericException Crash when displaying a banner while MediationDebugger or native screen is open.

### ISSUE THREAD

**Issue by soqman** [AppLovin/AppLovin-MAX-Unity-Plugin#354]
### MAX Plugin Version

6.4.2

### Unity Version

2022.3.21f

### Device/Platform Info

ios 17.4.1 iPhone 13 mini 13

### Current Behavior

Fatal Exception: NSGenericException
Unable to activate constraint with anchors <NSLayoutXAxisAnchor:0x30067fe80 "MAAdView:0x151b91410.centerX"> and <NSLayoutXAxisAnchor:0x300708d80 "UILayoutGuide:0x30274b720'UIViewSafeAreaLayoutGuide'.centerX"> because they have no common ancestor. Does the constraint or its anchors reference items in different view hierarchies? That's illegal.

### Expected Behavior

No crash

### How to Reproduce

1. Set banner is active,
2. open MediationDebugger
3. wait

or

1. Set banner is active
2. Show rate us ios native dialog
3. wait

**Comment by JonathanLiuApp** [comment]
@soqman We are aware of this issue and will have a fix in the next release. I will update here when the fix is live.

**Comment by abogarsukov-braingames** [comment]
We experience very similar crash after upgrading to 6.4.2. Attaching the crash report.
[applovin-ios-crash.zip](https://github.com/AppLovin/AppLovin-MAX-Unity-Plugin/files/14932679/applovin-ios-crash.zip)
In our case it randomly crashes when playing an ad.

**Comment by santoshbagadi** [comment]
We're working on releasing a fix for this shortly. If you wish to add the workaround now, you can do so by replacing [this](https://github.com/AppLovin/AppLovin-MAX-Unity-Plugin/blob/604c28373f4f272aeab3142143770fdb2d89f2a9/DemoApp/Assets/MaxSdk/AppLovin/Plugins/iOS/MAUnityAdManager.m#L1710) line in `Assets/MaxSdk/AppLovin/Plugins/iOS/MAUnityAdManager.m` file to the following 

```
UILayoutGuide *layoutGuide = superview.window ? superview.window.safeAreaLayoutGuide : superview.safeAreaLayoutGuide;
```

**Comment by wangping0214** [comment]
> We're working on releasing a fix for this shortly. If you wish to add the workaround now, you can do so by replacing [this](https://github.com/AppLovin/AppLovin-MAX-Unity-Plugin/blob/604c28373f4f272aeab3142143770fdb2d89f2a9/DemoApp/Assets/MaxSdk/AppLovin/Plugins/iOS/MAUnityAdManager.m#L1710) line in `Assets/MaxSdk/AppLovin/Plugins/iOS/MAUnityAdManager.m` file to the following
> 
> ```
> UILayoutGuide *layoutGuide = superview.window ? superview.window.safeAreaLayoutGuide : superview.safeAreaLayoutGuide;
> ```

Thank you, it works fine for me.

**Comment by JonathanLiuApp** [comment]
We've released a fix for this in the latest version of our plugin ([6.4.3](https://github.com/AppLovin/AppLovin-MAX-Unity-Plugin/releases/tag/release_6_4_3)).

**Comment by RiverboatGambler** [comment]
I'm seeing this error in XCode using 12.4.1

Is it truly fixed or is this a different error?

```
[AppLovinSdk] DEBUG [ALErrorReporter] storekit_error reported with code 200 and extra parameters {
    "ad_id" = 24628950;
    "ad_size" = INTER;
    "error_message" = "Error Domain=SKANErrorDomain Code=10 \"(null)\"";
    operation = start;
    parameters = "{\"ad_impression_id\":\"d6e94b48-5dfe-4c60-a165-860e50e51b13\",\"ad_network_id\":\"ludvb6z3bs.skadnetwork\",\"ts_s\":1714511968060,\"signature\":\"MDQCGCid/aXHF7cxeUp5OgALow2Q6aV+oN6lzQIYNuBWeY3Mb+zJJk1qbCZxZYn+C0wXI3pV\",\"advertised_app_store_item_id\":0,\"ad_campaign_id\":51,\"version\":\"3.0\",\"source_app_store_item_id\":0}";
    source = SKAdImpression;
}
```

**Comment by thomasmso** [comment]
@RiverboatGambler - Your error is different than the original issue. However, I am curious of your error message, are you able to easily reproduce that upon showing interstitial ads from AppLovin?

**Comment by RiverboatGambler** [comment]
Yes. It happens everytime I display a Test Reward Ad on a Simulator in XCode.  I don't know why AppLovin would care anything about storekit. I am only serving bid ads through AdMob.

<img width="1321" alt="Screenshot 2024-05-01 at 12 06 11 PM" src="https://github.com/AppLovin/AppLovin-MAX-Unity-Plugin/assets/59775247/57a2d845-d351-4972-a1bd-6d5d199be8fb">

**Comment by thomasmso** [comment]
@RiverboatGambler - Got it, yeah that's fine. It's an error that is isolated to simulators only. We'll remove the error logs in an upcoming SDK release, thanks for reporting it nonetheless and let us know if anything else arises.

**Comment by RiverboatGambler** [comment]
Glad to hear it. I just replied to a Help Desk ticket. I had to pull the latest SDK. I can't really provide any additional information because the app runs on a simulator without any log message that indicate an issue, but TestFlight crashes immediately on launch using an actual iPhone. The app crashes too quickly to get any Crashlytics information. 

All I can add is that to use version 12.4.1, I also had to upgrade the GoogleMobileAds package to 11.3.0 and Firebase to 10.24.0.  

Now, when I submit to Apple for review, I receive the following warning:

```
Although delivery was successful, you may want to correct the following issues in your next delivery. Once you've corrected the issues, upload a new binary to App Store Connect.

ITMS-90863: Macs with Apple silicon support issue - The app links with libraries that aren’t present in macOS:

@rpath/AppLovinSDK.framework/AppLovinSDK
Apple Developer Relations
```
It's worth noting that I've also updated the SDK for LiftOff/Vungle in the build as well.
